### PR TITLE
MN UP Receiver Migration Pre-Work

### PR DESCRIPTION
**Pull request by oslynn**
This PR MN UP Receiver migration pre-work

Test Steps:
1. *Include steps to test these changes*

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Review by JessicaWNava — CHANGES_REQUESTED** [review]
One question, after that is addressed, will approve. Looks good otherwise.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  PID: '/PATIENT_RESULT/PATIENT/PID'
+  SPM: '/PATIENT_RESULT/ORDER_OBSERVATION/SPECIMEN/SPM'
+
+elements:
+
+  - name: xtn-use-code-from-system
+    condition: '%resource.system = "email"'
+    value: [ '"NET"' ]
+    hl7Spec: [ '%{hl7TelecomPath}-2' ]
+
```
Comment (victor-chaparro) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml:14]: @oslynn Why are you include this for every STLT why is this needed?

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  - name: mn-message-encoding-characters
+    condition: 'true'
+    value: [ '"^~\\&"' ]
+    hl7Spec: [ 'MSH-2' ]
+
```
Comment (victor-chaparro) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml:19]: Same with this, why is this needed?

**Review by victor-chaparro — CHANGES_REQUESTED** [review]

File: prime-router/settings/STLTs/MN/mn-doh.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        whenEmpty:
+          action: "NONE"
+          onlyOncePerDay: false
+      description: "ELR Primary for MN DOH"
+      transport: !<SFTP>
+        host: "mdh-ftp.health.state.mn.us"
+        port: "22"
+        filePath: "./PROD"
+        credentialName: null
+        type: "SFTP"
```
Comment (victor-chaparro) [inline prime-router/settings/STLTs/MN/mn-doh.yml:97]: Remove production transport

**Review by victor-chaparro — CHANGES_REQUESTED** [review]

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml (hunk 1 above)
Comment (oslynn) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml:14]: Ok, if we don't need it, I will delete it.

**Review by victor-chaparro — APPROVED** [review]

File: prime-router/settings/STLTs/MN/mn-doh.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        - "(Bundle.entry.resource.ofType(ServiceRequest)[0].requester.resolve().organization.resolve().address.state = 'MN') or (Bundle.entry.resource.ofType(Patient).address.state = 'MN')"
+      qualityFilter:
+        - "((Bundle.entry.resource.ofType(Specimen).collection.collectedPeriod.exists() or Bundle.entry.resource.ofType(Specimen).collection.collected.exists()) or Bundle.entry.resource.ofType(serviceRequest).occurrence.exists() or Bundle.entry.resource.ofType(Observation).effective.exists())"
+        - "(Bundle.entry.resource.ofType(Patient).address.line.exists() or Bundle.entry.resource.ofType(Patient).address.postalCode.exists() or Bundle.entry.resource.ofType(Patient).telecom.exists())"
+        - "Bundle.entry.resource.ofType(MessageHeader).id.exists()"
+        - "Bundle.entry.resource.ofType(Patient).birthDate.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.family.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.given.count() > 0"
+        - "Bundle.entry.resource.ofType(Specimen).type.exists()"
+      routingFilter: []
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/MN/mn-doh.yml:73]: This routing filter was removed and needs to be moved over: - "matches(test_result_status, F, C, P)"

File: prime-router/settings/STLTs/MN/mn-doh.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  description: "Minnesota Department of Health"
+  jurisdiction: "STATE"
+  stateCode: "MN"
+  countyName: null
+  filters: null
+  featureFlags: null
+  keys: null
+  senders: []
+  receivers:
+    - name: "full-elr"
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/MN/mn-doh.yml:15]: There is a secondary active filter in their settings that needs to be moved over as well.

File: prime-router/settings/STLTs/MN/mn-doh.yml (hunk 4 above)
Comment (JessicaWNava) [inline prime-router/settings/STLTs/MN/mn-doh.yml:73]: Also missing routing filter - "DoesNotMatch(test_performed_code, 85478-6,76080-1,92141-1,80383-3,82170-2,85477-8,76078-5,77026-3,77028-9,77027-1,92142-9,80382-5,82166-0,82167-8,82169-4,82168-6,92809-3,92808-5,95423-0)"

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  - name: mn-patient-race-coding-system
+    resource: 'Bundle.entry.resource.ofType(Patient).extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.coding'
+    condition: '%resource.code.exists()'
+    value: [ '%resource.system.getCodingSystemMapping()' ]
+    hl7Spec: [ '%{PID}-10-3' ]
+
+  - name: mn-suppress-HL7-Fields-SPM-2-1
+    condition: 'true'
+    value: ['""']
```
Comment (JessicaWNava) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml:63]: Can you please add an integration test for this receiver as well?

**Review by JessicaWNava — CHANGES_REQUESTED** [review]
There are some things, including a secondary receiver, that were left out. Please address and let me know when it is ready for re-review. Thank you!

File: prime-router/settings/STLTs/MN/mn-doh.yml (hunk 5 above)
Comment (oslynn) [reply prime-router/settings/STLTs/MN/mn-doh.yml:15]: The second active doesn't being used.  There is no transport for it.

File: prime-router/settings/STLTs/MN/mn-doh.yml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        type: "HL7"
+        truncationConfig:
+          truncateHDNamespaceIds: false
+          truncateHl7Fields: []
+          customLengthHl7Fields: {}
+      jurisdictionalFilter:
+        - "(Bundle.entry.resource.ofType(ServiceRequest)[0].requester.resolve().organization.resolve().address.state = 'MN') or (Bundle.entry.resource.ofType(Patient).address.state = 'MN')"
+      qualityFilter:
+        - "((Bundle.entry.resource.ofType(Specimen).collection.collectedPeriod.exists() or Bundle.entry.resource.ofType(Specimen).collection.collected.exists()) or Bundle.entry.resource.ofType(serviceRequest).occurrence.exists() or Bundle.entry.resource.ofType(Observation).effective.exists())"
+        - "(Bundle.entry.resource.ofType(Patient).address.line.exists() or Bundle.entry.resource.ofType(Patient).address.postalCode.exists() or Bundle.entry.resource.ofType(Patient).telecom.exists())"
```
[the remark this answers is not in the document]
Comment (oslynn) [reply prime-router/settings/STLTs/MN/mn-doh.yml:67]: Which previous settings that you are referring to?  I am looking at their prod setting and didn't see those settings.  Would you let me know?

File: prime-router/settings/STLTs/MN/mn-doh.yml (hunk 4 above)
Comment (oslynn) [reply prime-router/settings/STLTs/MN/mn-doh.yml:73]: Thank you.

File: prime-router/settings/STLTs/MN/mn-doh.yml (hunk 5 above)
Comment (JessicaWNava) [reply prime-router/settings/STLTs/MN/mn-doh.yml:15]: Got it, I missed that, thank you

File: prime-router/settings/STLTs/MN/mn-doh.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (oslynn) [reply prime-router/settings/STLTs/MN/mn-doh.yml:67]: - "hasAtLeastOneOf(patient_street,patient_zip_code,patient_phone_number,patient_email)"
I think the following is good.  Because patient_email is in the .telecom resource portion:
- "(Bundle.entry.resource.ofType(Patient).address.line.exists() or Bundle.entry.resource.ofType(Patient).address.postalCode.exists() or Bundle.entry.resource.ofType(Patient).telecom.exists())"

[the remark this answers is not in the document]
Comment (JessicaWNava) [reply prime-router/settings/STLTs/MN/mn-doh.yml:67]: Ah, you are right. I forgot the email was included in the telecom. Thank you!

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml (hunk 6 above)
Comment (oslynn) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MN/MN-receiver-transform.yml:63]: We can add integration test when we do the real migration.  I think, it is not required for pre-work.

**Review by JessicaWNava — APPROVED** [review]
Looks good to me.
